### PR TITLE
Modify the extensions that show up in "Installed marketing extensions" card

### LINKED ISSUE #34431: Make new extensions show up in "Installed marketing extensions" card
- Part of epic issue https://github.com/woocommerce/woocommerce/issues/33895.
- Related to task issue https://github.com/woocommerce/woocommerce/issues/34107 and its PR https://github.com/woocommerce/woocommerce/pull/34415.

After users installed an extension from the "Recommended extensions" / "Discover more marketing tools" card, the extension should show up in the "Installed marketing extensions" card.

In PR https://github.com/woocommerce/woocommerce/pull/34415, we removed HubSpot, and added the following new extensions:

- [ ] AutomateWoo Referrals
- [ ] AutomateWoo Birthdays
- [ ] Creative Mail by Constant Contact
- [ ] Jetpack CRM
- [ ] Salesforce
- [ ] Tiktok
- [ ] Zapier

When we install these extensions, they do not show up in the "Installed marketing extensions" card. This issue is to make them show up.

We should also remove the related unused code for HubSpot.

Icons for the extensions are in the PR https://github.com/woocommerce/woocommerce/pull/34415, so PR for this issue should be based on that PR.

## Technical details

The place where we control whether an extension shows up or not is in:
 https://github.com/woocommerce/woocommerce/blob/41a276c49b47c63973a7c59abe38b165549de81d/plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php

We would need to look into each extensions and put in code specific to each extension to make them show up.

## Expected outcome

When a recommended marketing extension is installed, it should show up in the "Installed marketing extensions" card:

image

I think there is also a way to check via API, see: 

https://github.com/woocommerce/woocommerce/blob/41a276c49b47c63973a7c59abe38b165549de81d/plugins/woocommerce/src/Admin/API/MarketingOverview.php#L127-L129

I think the API is called in PHP and the response data is injected into HTML, and then JS is used to get it out from HTML and put it into wp-data store, and we have a related wp-data selector here:

https://github.com/woocommerce/woocommerce/blob/41a276c49b47c63973a7c59abe38b165549de81d/plugins/woocommerce-admin/client/marketing/data/selectors.js#L1-L3

### PR DESCRIPTION
**Pull request by nima-karimi**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #34431.

This PR adds seven new extensions to the "Installed marketing extensions" list on the `Marketing > Overview` page. These extensions appear on the list when the user installs and activates them. The added extensions are:

- [AutomateWoo Referrals](https://woocommerce.com/products/automatewoo-refer-a-friend/)
- [AutomateWoo Birthdays](https://woocommerce.com/products/automatewoo-birthdays/)
- [Creative Mail by Constant Contact](https://woocommerce.com/products/creative-mail-for-woocommerce/)
- [Jetpack CRM](https://woocommerce.com/products/jetpack-crm/)
- [Salesforce](https://woocommerce.com/products/integration-with-salesforce-for-woocommerce/)
- [Tiktok](https://woocommerce.com/products/tiktok-for-woocommerce/)
- [Zapier](https://woocommerce.com/products/woocommerce-zapier/)
- [Vimeo for WooCommerce](https://woocommerce.com/products/trustpilot-reviews/)
- [Trustpilot Reviews](https://woocommerce.com/products/trustpilot-reviews/)

Moreover, I have removed the [HubSpot](https://woocommerce.com/products/hubspot-for-woocommerce/) extension from the list based on the requirements defined in #34431.

This PR is based on Gan's PR #34415 and uses the extension icons introduced there.

Note: The `InstalledExtensions.php` file needed some cleanup/refactoring; however, I did not modify it because we will implement a new method for getting this data in the Multi-channel feature (pe2C5g-dl-p2) that will deprecate this class.

### How to test the changes in this Pull Request:

- General tests:

  1. Go to `WooCommerce Settings > Advanced > Features` and turn off the multichannel marketing feature.
  2. Install the seven extensions mentioned above
  3. Install AutomateWoo
  4. Go to the `Marketing > Overview` page and confirm their names appear in the "Installed marketing extensions" section
  5. Confirm that the "Get Support" and "Docs" links are correct and point to each extension's support page and documentation

- For [AutomateWoo Referrals](https://woocommerce.com/products/automatewoo-refer-a-friend/), [AutomateWoo Birthdays](https://woocommerce.com/products/automatewoo-birthdays/), [Jetpack CRM](https://woocommerce.com/products/jetpack-crm/), and [Zapier](https://woocommerce.com/products/woocommerce-zapier/):

  1. Confirm that the "Settings" link works and redirects the user to the extension's settings page.

- For [Creative Mail by Constant Contact](https://woocommerce.com/products/creative-mail-for-woocommerce/), [Salesforce](https://woocommerce.com/products/integration-with-salesforce-for-woocommerce/), [Vimeo for WooCommerce](https://woocommerce.com/products/trustpilot-reviews/), and [Tiktok](https://woocommerce.com/products/tiktok-for-woocommerce/):

  1. Ensure you have not set up the extension or completed onboarding. If you have, reset the extension data.
  2. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to their onboarding/setup process.
  3. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, i.e.` pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by nima-karimi** [comment]
Updated this PR to add [Vimeo for WooCommerce](https://woocommerce.com/products/trustpilot-reviews/) and [Trustpilot Reviews](https://woocommerce.com/products/trustpilot-reviews/) to the list of extensions on the marketing page.

File: plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php [hunk 1]
```diff
@@ -319,6 +377,200 @@ protected static function get_mailpoet_extension_data() {
...
+		$data['icon'] = WC_ADMIN_IMAGES_FOLDER_URL . '/marketing/tiktok.jpg';
+
+		if ( 'activated' === $data['status'] ) {
+			if ( false !== get_option( 'tt4b_access_token' ) ) {
+				$data['status'] = 'configured';
+			}
+
+			$data['settingsUrl'] = admin_url( 'admin.php?page=tiktok' );
+			$data['docsUrl']     = 'https://woocommerce.com/document/tiktok-for-woocommerce/';
+			$data['supportUrl']  = 'https://ads.tiktok.com/athena/user-feedback/?identify_key=6a1e079024806640c5e1e695d13db80949525168a052299b4970f9c99cb5ac78';
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php:436]: There is a system error when I click on the link. Is this expected?

https://user-images.githubusercontent.com/4344253/187343722-fb6c5b66-ac39-4a7c-bb98-b86ed1eb05b9.mov

**Comment by chihsuan** [comment]
> Confirm that the "Get Support" and "Docs" links are correct and point to each extension's support page and documentation

For other reviewers: The `multichannel-marketing` feature flag is enabled now. To test these links, I think we need to set `multichannel-marketing` to false in developement.json.

File: plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php [hunk 2]
```diff
@@ -319,6 +377,200 @@ protected static function get_mailpoet_extension_data() {
...
+
+
+		if ( 'activated' === $data['status'] && class_exists( '\Integration_With_Salesforce_Admin' ) ) {
+			if ( ! method_exists( '\Integration_With_Salesforce_Admin', 'get_connection_status' ) || \Integration_With_Salesforce_Admin::get_connection_status() ) {
+				$data['status'] = 'configured';
+			}
+
+			$data['settingsUrl'] = admin_url( 'admin.php?page=integration-with-salesforce' );
+			$data['docsUrl']     = 'https://woocommerce.com/document/salesforce-integration/';
+			$data['supportUrl']  = 'https://wpswings.com/submit-query/?utm_source=wpswings-salesforce-woo&utm_medium=woo-backend&utm_campaign=submit-query';
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php]: Is the query string needed?

 `?utm_source=wpswings-salesforce-woo&utm_medium=woo-backend&utm_campaign=submit-query`

Comment (nima-karimi) [reply plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php]: You're right, we can remove that tracking URL. Done in https://github.com/woocommerce/woocommerce/pull/34444/commits/19b91307bc6e70d441919bc20e500adc7c314132

File: plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php (hunk 1 above)
Comment (nima-karimi) [reply plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php:436]: Yes, that appears when you haven't logged in with your TikTok account. I agree that it's strange behavior and a bit confusing, but I'm using the same link as suggested in [TikTok's Documentation](https://woocommerce.com/document/tiktok-for-woocommerce/#section-9).

**Comment by chihsuan** [comment]
Hi @nima-karimi Thanks for addressing the feedback! 

I just tested your PR, and found that if we don't activate `AutomateWoo`, `AutomateWoo - Refer A Friend Add-on` shows an **invalid** "Finish setup" link. Could you confirm it?


https://user-images.githubusercontent.com/4344253/188067720-6a80af66-7e56-48e0-b6ff-182a2bbf3790.mov

**Comment by ecgan** [comment]
> For other reviewers: The `multichannel-marketing` feature flag is enabled now. To test these links, I think we need to set `multichannel-marketing` to false in developement.json.

Hi @chihsuan , the feature flag was developed in PR https://github.com/woocommerce/woocommerce/pull/33964. The flag is set to `true` based on https://github.com/woocommerce/woocommerce/blob/ca91250b2e17e88c902e460135c0531b3f632d90/plugins/woocommerce-admin/docs/features/feature-flags.md:

> Development - All features should be enabled in development.

As an alternative option to modifying the `development.json` file, you can also toggle the feature in the Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`. PR https://github.com/woocommerce/woocommerce/pull/33964 contains a demo video on this. 🙂

**Comment by nima-karimi** [comment]
> I just tested your PR, and found that if we don't activate AutomateWoo, AutomateWoo - Refer A Friend Add-on shows an invalid "Finish setup" link. Could you confirm it?

@chihsuan Ah, nice catch. This might be tricky because I don't think we can define dependencies between extensions on the marketing page. In this case, we can only show the links to their support page and documentation and remove the link to their Settings page (because that page won't exist if AW is disabled). I made some changes in https://github.com/woocommerce/woocommerce/pull/34444/commits/e67d490194e2e3915288adb3389bcfc095f9403c to apply this.

Alternatively, we can add a link to activate AutomateWoo next to those two extensions, but that would also require some work in the front end (cc: @ecgan).

**Review by chihsuan — APPROVED** [review]
Thanks @nima-karimi tested well and looks nice! 👍

**Pull request merged**